pkg/pillar/hypervisor/kvm.go

```diff
@@ -954,13 +955,11 @@ func (ctx KvmContext) CreateDomConfig(domainName string,
 	// PCI ID conflicts, let qemu assign PCI ID for vsock.
 	vsockContext := struct {
 		GuestCID string
-	}{GuestCID: fmt.Sprintf("%d", clientCid)}
+	}{GuestCID: fmt.Sprintf("%d", atomic.AddUint32(&clientCid, 1))}
```

Comment:
Can you please squash those to the first commit? Probably also makes sense to add a comment explaining why atomic is used .

Reply:
done.